src/ILCompiler/repro/Program.cs

```diff
@@ -7,7 +7,76 @@ internal class Program
 {
     private static void Main(string[] args)
     {
-        Console.WriteLine("Hello world");
+        MyInterface i1 = new Foo(false, 1, "Foo implementation") as MyInterface;
```

Comment:
I would keep the Program.cs in repro as is. Instead, we should add a new smoke test under tests\src\simple that has what you got here.